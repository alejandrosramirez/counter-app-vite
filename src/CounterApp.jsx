import { useState } from "react";
import PropTypes from "prop-types";

export const CounterApp = ({ value }) => {
	const [counter, setCounter] = useState(value);
	const handleAdd = () => setCounter(counter + 1);
	const handleSubstract = () => setCounter(counter - 1);
	const handleReset = () => setCounter(value);

	return (
		<>
			<h1>Counter App</h1>
			<h2>{counter}</h2>

			<button aria-label="btn-add" onClick={handleAdd}>+1</button>
			<button aria-label="btn-sub" onClick={handleSubstract}>-1</button>
			<button aria-label="btn-res" onClick={handleReset}>Reset</button>
		</>
	);
};

CounterApp.propTypes = {
	value: PropTypes.number.isRequired,
};
