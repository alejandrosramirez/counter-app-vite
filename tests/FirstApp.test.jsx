import { render } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe("Pruenas en <FirstApp />", () => {
	test("Debe hacer match con el snapshot", () => {
		const title = "Hola, soy el Kukun";
		const { container } = render(<FirstApp title={title} />);
		expect(container).toMatchSnapshot();
	});

	test("Debe mostrat el título en un <h1></h1>", () => {
		const title = "Hola, soy el Kukun";
		const { container, getByText, getByTestId } = render(
			<FirstApp title={title} />
		);
		expect(getByText(title)).toBeTruthy();
		// const h1 = container.querySelector("h1");
		// expect(h1.innerHTML).toBe(title);
		expect(getByTestId("test-title").innerHTML).toContain(title);
	});

	test("Debe de mostrar el título enviado por props", () => {
		const title = "Hola, soy el Kukun";
		const subtitle = "Holis";
		const { getByText } = render(
			<FirstApp
				title={title}
				subtitle={subtitle}
			/>
		);
		expect(getByText(subtitle)).toBeTruthy();
	});
});
