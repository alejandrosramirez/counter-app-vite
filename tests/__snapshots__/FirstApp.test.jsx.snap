// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruenas en <FirstApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy el Kukun
  </h1>
  <p>
    No hay subt√≠tulo
  </p>
  <p>
    Alex
  </p>
</div>
`;
